spec/lib/contextual_logger_spec.rb

```diff
@@ -115,6 +116,38 @@ def expect_log_line_to_be_written(log_line)
         expect(console_log_stream.string.gsub(/\[[^]]+\]/, '[]')).to eq("E, [] ERROR -- {:service=>\"test_service\"}: error message\n")
       end
     end
+
+    describe 'with nil message' do
+      it "logs null" do
+        logger.level = Logger::Severity::DEBUG
+        log_message_at_every_level(logger, nil)
+
+        expect(log_stream.string.gsub(/"timestamp":"[^"]+"/, '<time>')).to eq(<<~EOS)
+          {"message":null,"severity":"DEBUG",<time>}
+          {"message":null,"severity":"INFO",<time>}
+          {"message":null,"severity":"WARN",<time>}
+          {"message":null,"severity":"ERROR",<time>}
+          {"message":null,"severity":"FATAL",<time>}
+          {"message":null,"severity":"ANY",<time>}
+        EOS
+      end
+    end
+
+    describe 'with false message' do
+      it "logs false" do
+        logger.level = Logger::Severity::DEBUG
+        log_message_at_every_level(logger, false)
+
+        expect(log_stream.string.gsub(/"timestamp":"[^"]+"/, '"timestamp":"<time>"')).to eq(<<~EOS)
+          {"message":false,"severity":"DEBUG","timestamp":"<time>"}
+          {"message":false,"severity":"INFO","timestamp":"<time>"}
+          {"message":false,"severity":"WARN","timestamp":"<time>"}
+          {"message":false,"severity":"ERROR","timestamp":"<time>"}
+          {"message":false,"severity":"FATAL","timestamp":"<time>"}
+          {"message":false,"severity":"ANY","timestamp":"<time>"}
```

Comment:
Same point here about non-strings. You can see it's `"message":false` where to be consistent with `:Logger`, we might want `"message":"false"`.

Reply:
Read other thread